ntoskrnl/io/iomgr/deviface.c

```diff
@@ -28,6 +29,257 @@ IopGetDeviceObjectFromDeviceInstance(PUNICODE_STRING DeviceInstance);
 
 static PWCHAR BaseKeyString = L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\DeviceClasses\\";
 
+/**
+ * @brief IopBuildSymbolicLink
+ * 
+ * Creates a new symbolic link from the specified prefix, device string,
+ * class GUID and reference string (if any).
+ * 
+ * @param[in] PrefixString
+ * Prefix of symbolic link. '\??\' for Kernel mode or '\\?\' for User mode.
+ * 
+ * @param[in] DeviceString
+ * Device string, placed after prefix and before GUID, looks like ACPI#PNP0501#1#.
+ * 
+ * @param[in] GuidString
+ * Device interface class GUID represented by a string. Placed in curly brackets {},
+ * after device string, should always be 38 characters long. Looks like
+ * {01234567-89ab-cdef-0123-456789abcdef}.
+ * 
+ * @param[in,opt] ReferenceString
```

Comment:
```suggestion
 * @param[in] ReferenceString
```
"opt" is an unknown doxygen keyword (sadly) and this will break it.
See for example:
https://doxygen.reactos.org/d6/d1e/user_2winsrv_2concfg_2font_8h.html#aced062bf0f2e4268480ea28d4dd495b5
![image](https://github.com/reactos/reactos/assets/1969829/3c831d99-5ea7-46d5-9854-514e1c0f3975)

Reply:
Fixed, didn't know that. Thanks. Changed this for separate function also.